embrace-android-core/src/main/kotlin/io/embrace/android/embracesdk/internal/delivery/caching/PayloadCachingServiceImpl.kt

```diff
@@ -0,0 +1,28 @@
+package io.embrace.android.embracesdk.internal.delivery.caching
+
+import io.embrace.android.embracesdk.internal.payload.Envelope
+import io.embrace.android.embracesdk.internal.payload.SessionPayload
+import io.embrace.android.embracesdk.internal.session.caching.PeriodicBackgroundActivityCacher
+import io.embrace.android.embracesdk.internal.session.caching.PeriodicSessionCacher
+
+class PayloadCachingServiceImpl(
+    private val periodicSessionCacher: PeriodicSessionCacher,
+    private val periodicBackgroundActivityCacher: PeriodicBackgroundActivityCacher
+) : PayloadCachingService {
+
+    override fun shutdown() {
+    }
+
+    override fun startCaching(isInBackground: Boolean, supplier: () -> Envelope<SessionPayload>?) {
+        if (isInBackground) {
+            periodicBackgroundActivityCacher.scheduleSave(supplier)
+        } else {
+            periodicSessionCacher.start(supplier)
+        }
+    }
+
+    override fun stopCaching() {
```

Comment:
This seems pretty haphazard, not knowing what we are exactly stopping. Perhaps we should be explicit about which session we are stopping caching for?